src/hardware/gameblaster.cpp

```diff
@@ -133,110 +131,108 @@ void GameBlaster::Open(const int port_choice, const std::string &card_choice,
 	is_open = true;
 }
 
-bool GameBlaster::RenderOnce()
+bool GameBlaster::MaybeRenderFrame(AudioFrame &frame)
 {
 	// Static containers setup once and reused
-	constexpr auto num_channels = 2; // left and right
-	static std::vector<int16_t> frame(num_channels);
-	static int16_t *buffer[] = {&frame[0], &frame[1]};
-	static std::vector<float> accumulator(num_channels);
+	static std::vector<int16_t> buf(2); // left and right
```

Comment:
Not that it matter a lot, but why not just use `std::array`? We know the size upfront.